packages/eslint-config-functional/index.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * strvcom/eslint-config-functional
+ *
+ * @author      Michal Jarnot <m.jarnot@yahoo.com>
+ * @copyright   2020 STRV
+ * @license     http://choosealicense.com/licenses/bsd-3-clause  BSD-3-Clause License
+ */
+
+'use strict'
+
+module.exports = {
+  plugins: ['functional'],
+  rules: {
+    //
+    // No Mutations Rules
+    //
+
+    'functional/immutable-data': 'error',
+    'functional/no-let': 'error',
+    'functional/no-method-signature': 'error',
+    'functional/prefer-readonly-type': 0,
+
+    //
+    // No Object-Orientation Rules
+    //
+
+    'functional/no-class': 'error',
+    'functional/no-mixed-type': 'error',
+    'functional/no-this-expression': 'error',
+    'functional/prefer-type-literal': 0,
+
+    //
+    // No Statements Rules
+    //
+
+    'functional/no-conditional-statement': 0,
+    'functional/no-expression-statement': 'error',
+    'functional/no-loop-statement': 'error',
+    'functional/no-return-void': 0,
+
+    //
+    // No Exceptions Rules
+    //
+
+    'functional/no-promise-reject': 0,
+    'functional/no-throw-statement': 0,
+    'functional/no-try-statement': 0,
+
+    //
+    // Currying Rules
+    //
+
+    'functional/functional-parameters': 'error',
```

Comment:
Okay, I actually disagree with all of these. 😂

They could be nice as an optional addition to the codebase, but it's rarely just black & white, functional & object-oriented.
This doesn't really enforce good code practices but only limits your toolset.

Reply:
Actually I don't use this config anywhere yet. `no-loop-statement` is also problematic. Even though this config is not as strict as it could be it is intended to help create purely functional codebase. 🙂 

I would love to hear more opinions about this though. Maybe we can make it useful for FE React codebase. I don't think this will ever be usable on BE for me because of how Nest.js works.